### PR TITLE
feat(debugging): add nvim-dap-repl-highlights

### PR DESCRIPTION
**Pull request by tku137**
Add [nvim-dap-repl-highlights](https://github.com/LiadOz/nvim-dap-repl-highlights) plugin. Tree-sitter setup needs to be called after plugin setup or `dap_repl` language won't be found. No `opts` for plugin setup, as far as I can see.

### CODE REVIEW

File: lua/astrocommunity/debugging/nvim-dap-repl-highlights/init.lua [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  config = function()
+    require("nvim-dap-repl-highlights").setup()
+    require("nvim-treesitter.configs").setup {
+      highlight = {
+        enable = true,
+      },
+      ensure_installed = { "dap_repl" },
+    }
+  end,
+}
```
Comment (mehalter) [inline lua/astrocommunity/debugging/nvim-dap-repl-highlights/init.lua:14]:
```suggestion
local utils = require "astronvim.utils"
return {
  "nvim-treesitter/nvim-treesitter",
  dependencies = {
    "LiadOz/nvim-dap-repl-highlights",
    dependencies = { "mfussenegger/nvim-dap" },
    opts = {},
  },
  opts = function(_, opts)
    if opts.ensure_installed ~= "all" then
      opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, "dap_repl")
    end
  end,
}
```

**Review by mehalter — CHANGES_REQUESTED** [review]
A bit of a restructure recommendations

**Review by mehalter — APPROVED** [review]
LGTM!

**Comment by tku137** [comment]
Out of curiosity: the restructuring enables the use of `list_insert_unique()` on the `ensure_installed` list of tree sitter? Did I get that right?

**Comment by mehalter** [comment]
Yeah just to make sure it's type safe and also provides the options that are being set in a way that the end user can override them if they need to. This also allows us to leverage the built in support for lazy.nvim to call setup rather than us specifying it manually

**Pull request merged**